src/test/ui/consts/recursive-zst-static.rs

```diff
@@ -1,5 +1,4 @@
-// revisions: default unleash
-//[unleash]compile-flags: -Zunleash-the-miri-inside-of-you
+// revisions: default
```

Comment:
This basically reverts https://github.com/rust-lang/rust/pull/71604. I specifically wrote that PR to make sure that the dynamic checks are able to catch this problem, even if the static checks might one day evolve to also be able to handle it (e.g. via https://github.com/rust-lang/rust/pull/71526). So, this change potentially reduces future test coverage for the dynamic checks, which is not great.

That's why I think "`-Zmiri-unleash` will make compilation fail" is a better safety net than your proposal. It catches `read_from_static.rs` (which we all agree should not use the flag), but permits this test here.

Reply:
Yea, this issue is why I left a longer comment in the PR's main comment, even though I misexplained what is going on exactly in the opposite way.